src/torchmetrics/image/qnr.py

```diff
@@ -0,0 +1,225 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.image.d_lambda import _spectral_distortion_index_compute, _spectral_distortion_index_update
+from torchmetrics.functional.image.d_s import _spatial_distortion_index_compute, _spatial_distortion_index_update
+from torchmetrics.metric import Metric
+from torchmetrics.utilities import rank_zero_warn
+from torchmetrics.utilities.data import dim_zero_cat
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _TORCHVISION_AVAILABLE
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
+
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["QualityWithNoReference.plot"]
+
+if not _TORCHVISION_AVAILABLE:
+    __doctest_skip__ = ["QualityWithNoReference", "QualityWithNoReference.plot"]
+
+
+class QualityWithNoReference(Metric):
+    """Compute Quality with No Reference (QualityWithNoReference_) also now as QNR.
+
+    The metric is used to compare the joint spectral and spatial distortion between two images.
+
+    As input to ``forward`` and ``update`` the metric accepts the following input
+
+    - ``preds`` (:class:`~torch.Tensor`): High resolution multispectral image of shape ``(N,C,H,W)``.
+    - ``target`` (:class:`~Dict`): A dictionary containing the following keys:
+        - ``ms`` (:class:`~torch.Tensor`): Low resolution multispectral image of shape ``(N,C,H',W')``.
+        - ``pan`` (:class:`~torch.Tensor`): High resolution panchromatic image of shape ``(N,C,H,W)``.
+        - ``pan_lr`` (:class:`~torch.Tensor`): Low resolution panchromatic image of shape ``(N,C,H',W')``.
```

Comment:
why is `pan_lr` required here but not in the functional implementation?

Reply:
`pan_lr` is optional in both. should I change the type of `pan_lr` to `Optional[torch.Tensor]`?